Users/aedler/operation mode patch (#177)
fiskaltrust/interface-doc: doc/general/operation-modes/operation-modes.md

### On-premise & off-premise installed components

> The on-premise solution of the fiskaltrust.Middleware requires [installation](../installation/operation-modes.md)  and configuration for the client. Below the most important components of the Middleware are illustrated:

Reviewer: ```suggestion
The on-premise solution of the fiskaltrust.Middleware requires [installation](../installation/operation-modes.md)  and configuration for the client. Below the most important components of the fiskaltrust.Middleware are illustrated:
```

![middleware-en](images/middleware-en.svg)

> #### IPOS Interface

Reviewer: ```suggestion
#### iPOS Interface
```

> The IPOS interface is part of the queue, a component of the fiskaltrust.Middleware. It enables the communication between the cash register and the fiskaltrust.Middleware and is identical for all supported countries (cross national).

Reviewer: ```suggestion
The iPOS interface is part of the queue, a component of the fiskaltrust.Middleware. It enables the communication between the cash register and the fiskaltrust.Middleware and is identical for all supported countries (cross national).
```

The iPOS Interface offers three interface methods:

- *echo* (check availability), 
- *sign* (sign receipt data, send special receipts) and 
- *journal* (export data).

> To establish a communication with the IPOS interface, following communication services are recommended based on the POS systems prerequisites:

Reviewer: ```suggestion
To establish a communication with the iPOS interface, following communication services are recommended based on the POS systems prerequisites:
```

| POS system technology stack | gRPC            | REST                       | WCF  |
| --------------------------- | --------------- | -------------------------- | ---- |
| **.NET client**             | **recommended** | alternative recommendation |      |
| **Java client**             | **recommended** |                            |      |
| **Web App**                 |                 | **recommended**            |      |

The availability of supported communication services is dependent on the market as shown in the following table:

| Communication service | AT            | DE            | FR            |
| --------------------- | ------------- | ------------- | ------------- |
| **gRPC**              | not supported | **supported** | not supported |
| **REST**              | **supported** | **supported** | **supported** |
| **WCF**               | **supported** | **supported** | **supported** |

> More detailed information, including the data flow, you can find in the [communication chapter](../communication/communication.md).

Reviewer: ```suggestion
More detailed information, including the data flow, please refer to the [communication chapter](../communication/communication.md).
```

#### Queue

> In addition to enable the communication with the cash register via IPOS Interface, the queue also serves to encapsulate the functionality of a receipt chain: All regular receipts created by the POS-System and sent to the fiskaltrust.Middleware are processed in the ft.SecurityMechanism. This mechanism takes care of the creation of the unique, consecutive receipt number, the chaining and the persistence of the data.  After getting secured and stored in the queue, the response of the fiskaltrust.SecurityMechanism is sent back to the POS system. All special receipts (for example the periodical closings) are sent as "requests to execute a special function" to the fiskaltrust.Middleware and get answered by it. 

Reviewer: ```suggestion
In addition to enable the communication with the cash register via iPOS Interface, the queue also serves to encapsulate the functionality of a receipt chain: All regular receipts created by the POS-System and sent to the fiskaltrust.Middleware are processed in the ft.SecurityMechanism. This mechanism takes care of the creation of the unique, consecutive receipt number, the chaining and the persistence of the data.  After getting secured and stored in the queue, the response of the fiskaltrust.SecurityMechanism is sent back to the POS-System. All special receipts (for example the periodical closings) are sent as "requests to execute a special function" to the fiskaltrust.Middleware and get responded by it. 
```

For detailed information on supported platforms and its restrictions, please refer to the linked appendices of the applicable markets in the table below:

| Platform        | AT            | DE                                                           | FR            |
| --------------- | ------------- | ------------------------------------------------------------ | ------------- |
| **Android**     | not supported | [**supported**](../../appendix-de-kassensichv/operation-modes/on-premise-platforms/android.md) | not supported |
| **Linux/macOs** | **supported** | [**supported**](../../appendix-de-kassensichv/operation-modes/on-premise-platforms/linux.md) | **supported** |
| **Windows**     | **supported** | [**supported**](../../appendix-de-kassensichv/operation-modes/on-premise-platforms/windows.md) | **supported** |

#### Launcher

> The Launcher is a software (file) named `fiskaltrust.exe`, which is used only for the on-premise installed products (e.g. AT products fiskaltrust.SignatureCard or fiskaltrust.SignatureBox). For Windows, it is a .NET command-line application and a .NET Windows service. For Linux and Mac, the launcher can be executed via Mono, version 3.2.8 or higher, or used as daemon.

Reviewer: ```suggestion
The Launcher is a software (file) named `fiskaltrust.exe`, which is used only for the on-premise installed products (e.g. AT product fiskaltrust.SignatureCloud). For Windows, it is a .NET command-line application and a .NET Windows service. For Linux and Mac, the launcher can be executed via Mono, version 3.2.8 or higher, or used as daemon.
```

The main tasks of the launcher are:

  - providing basic configuration settings such as ftCashBoxId and access token
  - comparing the configuration data retrieved from fiskaltrust.Helipad with the local configuration
  - updating queue- and SCU packages accordingly to the configuration
  - execution of configured packages
  - load balancing of multiple queues

> The executable file `fiskaltrust.exe` and the corresponding DLLs can be distributed via copy-paste and then configured and installed with the help of a command-line parameter. It can be downloaded (incl. configuration) from the Portal’s configuration->cashbox page, or found on nuget.org and configured manually.

Reviewer: ```suggestion
The executable file `fiskaltrust.exe` and the corresponding DLLs can be distributed via copy-paste and then configured and installed with the help of a command-line parameter. The launcher can be downloaded (incl. configuration) from the [https://portal.fiskaltrust.de/CashBox](https://portal.fiskaltrust.de/CashBox), or found on nuget.org and configured manually.
```

Call parameter of the launcher are documented in the chapter [installation](../installation/installation.md).

### Private cloud (operated by a third party) installed components

> No installation or configuration is required for the client and any platform can use this service. The availability and use of SaaS installed components is dependent on local regulations and currently available for Austria and Germany. In Germany, because of different market regulation, the fiskaltrust.Middleware components must be hosted and operated by the POS creator in the same datatcenter where the receipt generation process of a distributed point of sale system occurs.

Reviewer: ```suggestion
No installation or configuration is required for the client and any platform can use this service. The availability and use of SaaS installed components is dependent on local regulations and currently available for Austria and Germany. In Germany, because of different market regulation, the fiskaltrust.Middleware components must be hosted and operated by the PosCreator in the same datacenter where the receipt generation process of a distributed point of sale system occurs.
```

> The components for the SaaS solution are the same as for the local solution; except the launcher, which is not needed. In Germany, the fiskaltrust.Middleware components can be delivered by a predefined Kubernetes Namespace which can be deployed by a Helm-Chart. fiskaltrust provides a "Backend POD" Docker image and Helm-Charts to be deployed at the POS creators environment.

Reviewer: ```suggestion
The components for the SaaS solution are the same as for the local solution; except the launcher, which is not needed. In Germany, the fiskaltrust.Middleware components can be delivered by a predefined Kubernetes Namespace which can be deployed by a Helm-Chart. fiskaltrust provides a "Backend POD" Docker image and Helm-Charts to be deployed at the PosCreators environment.
```
